src/tools/immutable_object_cache/ObjectCacheStore.cc

```diff
@@ -97,6 +126,13 @@ int ObjectCacheStore::do_promote(std::string pool_nspace,
                    << " from pool id: " << pool_id
                    << " namespace: " << pool_nspace
                    << " snapshot: " << snap_id << dendl;
+  // try to take one token from TokenBucketThrottle
+  //   if success, directly return.
+  //   if fails, block here until new token be inserted bucket.
+  // TODO : because RO daemon just have one thread, it will block here..
```

Comment:
Let's not block here -- if we don't have any budget to promote, just return `OBJ_CACHE_SKIP`

Reply:
done. thanks